platforms/core-runtime/classloaders/src/main/java/org/gradle/internal/classloader/TransformErrorHandler.java

```diff
@@ -117,7 +117,8 @@ public ClassNotFoundException exitClassLoadingScopeWithException(Throwable th) t
         if (pendingException != null) {
             addSuppressedIfAvailable(th, pendingException);
         }
-        Throwables.propagateIfPossible(th, ClassNotFoundException.class);
+        Throwables.throwIfInstanceOf(th, ClassNotFoundException.class);
+        Throwables.throwIfUnchecked(th);
         throw new RuntimeException("Unexpected exception type", th);
```

Comment:
Can we replace these two lines with `UncheckedException.throwAsUncheckedException`? It already does the check if the exception is unchecked already, and wraps it if not (among other things)

Reply:
Done.